src/Uno.UI/UI/Xaml/Controls/TextBox/SinglelineTextBoxDelegate.iOS.cs

```diff
@@ -39,8 +39,16 @@ public override bool ShouldChangeCharacters(UITextField textField, NSRange range
 				{
 					var newLength = textBoxView.Text.Length + replacementString.Length - range.Length;
 					return newLength <= _textBox.GetTarget()?.MaxLength;
```

Comment:
I don't know, but this feels like a bug.

If `MaxLength` is specified (> 0), and `newLength < MaxLength`, then we are going to return true here. However, we should be returning false if the TextBox is ReadOnly.

Reply:
To be clear here, I'm commenting on an *existing* code. But I'm expecting the bug to be introduced in this PR specifically. This is because, previously, this return was the last return in the method, no further checks are done. But now, this is not the case.

Reply:
Thanks for the clarification, I misunderstood the comment the first time I read it, now I know what you meant.